android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java

```diff
@@ -68,6 +73,10 @@ public void onActivityResult(Activity activity, int requestCode, int resultCode,
 				if (promise != null) {
 					onShowActivityResult(resultCode, data, promise);
 				}
+			} else if (requestCode == PICK_DIR_REQUEST_CODE) {
```

Comment:
the body of this if is the same as above, you can likely use the previous `if` and change the condition to `requestCode == READ_REQUEST_CODE || requestCode == PICK_DIR_REQUEST_CODE)`

Reply:
I'm not sure I follow, the body is not the same - `onShowActivityResult(resultCode, data, promise)` vs `onPickDirectoryResult(resultCode, data, promise);`

Reply:
ah, crap, my mistake 😆

Reply:
still, maybe we can reduce the `promise != null` condition: 

```
if (promise == null) {
  Log.e(...)
  return;
}
```

Reply:
I'll do this but without the `Log.e` - I assume this callback may be called when some other activity returns so the fact that the promise is null isn't necessarily an issue.